File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/model/WorkerId.java
package ru.fix.distributed.job.manager.model;

import org.jetbrains.annotations.NotNull;
import ru.fix.distributed.job.manager.IdentityValidator;

import java.util.Arrays;
import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;

public class WorkerId implements Comparable<WorkerId> {

    private final String id;

    public WorkerId(String id) {
        IdentityValidator.validate(IdentityValidator.IdentityType.WorkerId, id);
        this.id = id;
    }

    public static Set<WorkerId> setOf(String... ids) {
        return Arrays.stream(ids).map(WorkerId::new).collect(Collectors.toSet());
    }

    public String getId() {
        return id;
    }

    @Override
    public int compareTo(@NotNull WorkerId o) {
        Objects.requireNonNull(o);
        return this.id.compareTo(o.getId());
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        WorkerId that = (WorkerId) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return "Worker[" + id + "]";
    }
}

File: distributed-job-manager/src/main/kotlin/ru/fix/distributed/job/manager/model/WorkerId.kt
package ru.fix.distributed.job.manager.model

data class WorkerId(val id: String) : Comparable<WorkerId> {

    override fun compareTo(other: WorkerId): Int {
        return id.compareTo(other.id)
    }

    override fun toString(): String {
        return "Worker[$id]"
    }

    companion object {
        private val PATTERN = "[a-zA-Z0-9_-[.]]+".toRegex()

        fun setOf(vararg ids: String): Set<WorkerId> {
            return ids.map { WorkerId(it) }.toSet()
        }
    }
}
